grafana/resource_alerting_notification_policy.go

```diff
@@ -53,12 +54,112 @@ func ResourceNotificationPolicy() *schema.Resource {
 				Optional:    true,
 				Description: "Minimum time interval for re-sending a notification if an alert is still firing. Default is 4 hours.",
 			},
+
+			"policy": {
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "Routing rules for specific label sets.",
+				Elem:        policySchema(supportedPolicyTreeDepth),
+			},
 		},
 	}
 }
 
+// The maximum depth of policy tree that the provider supports, as Terraform does not allow for infinitely recursive schemas.
+// This can be increased without breaking backwards compatibility.
+const supportedPolicyTreeDepth = 2
```

Comment:
I have a hunch (but not data) that 2 might be a bit low. I would start with something like 4.